lib/ansible/plugins/action/reboot.py

```diff
@@ -265,6 +311,7 @@ def validate_reboot(self):
     def run(self, tmp=None, task_vars=None):
         self._supports_check_mode = True
         self._supports_async = True
+        self._task_vars = task_vars
```

Comment:
This never feels right to me.  If we think of task_vars as being global to the instance, then it should be set externally, either in the constructor or by the caller setting it explicitly.  If we think of task_vars as being part of this method, then we shouldn't be setting the instance variable with it.

Reply:
I revamped everything to use local variables rather than storing `self._task_vars` as a class attribute.